test/test_jit_forms.py

```diff
@@ -1074,3 +1074,76 @@ def test_integral_grouping(compile_args):
         ]
     )
     assert len(unique_integrals) == 2
+
+
+@pytest.mark.parametrize("dtype", ["float64"])
+@pytest.mark.parametrize("permutation", [[0], [1]])
+def test_mixed_dim_form(compile_args, dtype, permutation):
+    # TODO Test 3D and non-simplex
+
+    ele_type = "Lagrange"
+    V_cell_type = "triangle"
+    Vbar_cell_type = "interval"
+
+    V_ele = basix.ufl.element(ele_type, V_cell_type, 2)
+    Vbar_ele = basix.ufl.element(ele_type, Vbar_cell_type, 1)
+
+    gdim = 2
+    V_domain = ufl.Mesh(basix.ufl.element("Lagrange", V_cell_type, 1, shape=(gdim,)))
+    Vbar_domain = ufl.Mesh(basix.ufl.element("Lagrange", Vbar_cell_type, 1, shape=(gdim,)))
+
+    V = ufl.FunctionSpace(V_domain, V_ele)
+    Vbar = ufl.FunctionSpace(Vbar_domain, Vbar_ele)
+
+    u = ufl.TrialFunction(V)
+    vbar = ufl.TestFunction(Vbar)
+
+    f = ufl.Coefficient(V)
+    g = ufl.Coefficient(Vbar)
+
+    ds = ufl.Measure("ds", domain=V_domain)
+
+    n = ufl.FacetNormal(V_domain)
+    forms = [ufl.inner(f * g * ufl.grad(u), n * vbar) * ds]
+    compiled_forms, module, code = ffcx.codegeneration.jit.compile_forms(
+        forms, options={"scalar_type": dtype}, cffi_extra_compile_args=compile_args
+    )
+    form0 = compiled_forms[0]
+    default_integral = form0.form_integrals[0]
+    kernel = getattr(default_integral, f"tabulate_tensor_{dtype}")
+
+    A = np.zeros((2, 6), dtype=dtype)
+    # FIXME Use more complex data
+    w = np.array([1, 1, 1, 1, 1, 1, 1, 1], dtype=dtype)
+    c = np.array([], dtype=dtype)
+    facet = np.array([0], dtype=np.intc)
+    perm = np.array(permutation, dtype=np.uint8)
+
+    xdtype = dtype_to_scalar_dtype(dtype)
+    coords = np.array([[0.0, 0.0, 0.0], [1.0, 0.0, 0.0], [0.0, 1.0, 0.0]], dtype=xdtype)
+
+    c_type = dtype_to_c_type(dtype)
+    c_xtype = dtype_to_c_type(xdtype)
+
+    ffi = module.ffi
+    kernel(
+        ffi.cast(f"{c_type}  *", A.ctypes.data),
+        ffi.cast(f"{c_type}  *", w.ctypes.data),
+        ffi.cast(f"{c_type}  *", c.ctypes.data),
+        ffi.cast(f"{c_xtype} *", coords.ctypes.data),
+        ffi.cast("int *", facet.ctypes.data),
+        ffi.cast("uint8_t *", perm.ctypes.data),
+    )
+
+    expected_result = np.array(
```

Comment:
It would be good to have some explanation as to what we are integrating here, and why this is the correct result.
I guess what you have is the facet normal on facet 0 (which is the one spanning (1.0,0) to (0.0, 1), i.e. a normal of (0.5, 0.5) multiplied by the gradient of the trial function on the parent cell.  It seems like both f and g are one (when looking at your coefficients, and it is not clear what order they are packed in). Could we use non-zero data here? 
This integral seems to simplify to `int_facet grad(u)[0]v + grad(u)[1]v * ds `